### PR TITLE
Link libdl in core/platform/default/stacktrace

### LINKED ISSUE #45013: Test linking fails due to missing ldl
**System information**
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.4.0-rc2

**Describe the problem**

Building the tests `//tensorflow/python/...` fails with a link error on `//tensorflow/python/tools:aot_compiled_test` due to stacktrace.h using `dladdr` but nothing in the chain links against `-ldl`.

Chain: aot_compiled_test -> "//tensorflow/core:test_main" -> "//tensorflow/core/platform:test_main" -> ":stacktrace_handler"

**Any other info / logs**
Looks like the issue can simply be fixed by adding the ldl flag to the ls test_main rule

### PR DESCRIPTION
**Pull request by Flamefire**
~Required due to use of `dladdr` via the `stacktrace_handler` dependency~

Required due to use of `dladdr` in `stacktrace.h` which is e.g. a dependency of `test_main` via `stacktrace_handler`

Fixes #45013

### CODE REVIEW

**Review by angerson — APPROVED** [review]

**Comment by Flamefire** [comment]
Seems CI wants a slightly different formatting. Added that change (only), the other failures are unrelated

**Review by learning-to-play — APPROVED** [review]

File: tensorflow/core/platform/BUILD [hunk 1]
```diff
@@ -1095,7 +1095,10 @@ cc_library(
     copts = tf_copts(),
     linkopts = select({
         "//tensorflow:windows": [],
-        "//conditions:default": ["-lm"],
+        "//conditions:default": [
+            "-lm",
+            "-ldl",
```
Comment (cantonios) [inline tensorflow/core/platform/BUILD:1100]: The flag should probably be added to linkopts here instead: https://github.com/tensorflow/tensorflow/blob/71738a2bfb3824611debc6c54dc26c50072473a0/tensorflow/core/platform/default/BUILD#L376

**Review by cantonios — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@Flamefire Can you please check @cantonios's comments and keep us posted ? Thank you!

**Comment by Flamefire** [comment]
Testing that now, thanks!

File: tensorflow/core/platform/BUILD (hunk 1 above)
Comment (Flamefire) [reply tensorflow/core/platform/BUILD]: Makes sense to me too. Added a commit doing that. Quick test shows that this works.

**Comment by Flamefire** [comment]
@gbaned Tested and rebased to implement the suggested change and resolve a conflict that occurred in the meantime.

**Review by cantonios — APPROVED** [review]

**Review by cantonios — APPROVED** [review]

**Comment by Flamefire** [comment]
Unrelated CI failures, e.g.:

> ERROR: The project you're trying to build requires Bazel 5.3.0 (specified in /workspace/.bazelversion), but it wasn't found in /usr/local/lib/bazel/bin.

**Pull request merged**